src/khoj/processor/org_mode/orgnode.py

```diff
@@ -126,6 +129,17 @@ def makelist(file, filename):
                     for parsedtag in parsedtags.split(":"):
                         if parsedtag != "":
                             tags.append(parsedtag)
+
+            # Add previous heading to ancestors if current heading is deeper than previous level
```

Comment:
Neat! This will make it a lot more useful, and hopefully will make it easier to retrieve sibling data in a given query.

Reply:
Yeah, this should improve search quality for nested org-mode entry trees by providing relevant, additional context